src/batch-transaction.js

```diff
@@ -0,0 +1,343 @@
+ * @param {object} apiResponse The full API response.
+ */
+/**
+ * Creates a set of read partitions that can be used to execute a read
+ * operation in parallel. Partitions become invalid when the transaction used
+ * to create them is closed.
+ *
+ * @param {ReadRequestOptions} options Configuration object, describing what to
+ *     read from.
+ * @param {CreateReadPartitionsCallback} [callback] Callback function.
+ * @returns {Promise<CreateReadPartitionsResponse>}
+ *
+ * @example <caption>include:samples/batch.js</caption>
+ * region_tag:create_read_partitions
+ */
+BatchTransaction.prototype.createReadPartitions = function(options, callback) {
+  var reqOpts = codec.encodeRead(options);
+  var gaxOpts = options.gaxOptions;
+
+  if (gaxOpts) {
+    delete reqOpts.gaxOptions;
+  }
+
+  this.createPartitions_(
+    {
+      client: 'SpannerClient',
+      method: 'partitionRead',
+      reqOpts,
+      gaxOpts,
+    },
+    callback
+  );
+};
+
+/**
+ * Executes partition.
+ *
+ * @see {@link Transaction#read} when using {@link ReadPartition}.
+ * @see {@link Transaction#run} when using {@link QueryParition}.
+ *
+ * @param {ReadPartition|QueryParition} partition The partition object.
+ * @param {TransactionRequestReadCallback|RunCallback} callback Callback
+ *     function.
+ * @returns {Promise<RunResponse>|Promise<TransactionRequestReadResponse>}
+ *
+ * @example <caption>include:samples/batch.js</caption>
+ * region_tag:execute_partition
+ */
+BatchTransaction.prototype.execute = function(partition, callback) {
+  if (is.string(partition.table)) {
+    this.read(partition.table, partition, callback);
+    return;
+  }
+
+  this.run(partition, callback);
+};
+
+/**
+ * Executes partition in streaming mode.
+ *
+ * @see {@link Transaction#createReadStream} when using {@link ReadPartition}.
+ * @see {@link Transaction#runStream} when using {@link QueryParition}.
+ *
+ * @param {ReadPartition|QueryPartition} partition The partition object.
+ * @returns {ReadableStream} A readable stream that emits rows.
+ *
+ * @example
+ * const Spanner = require('@google-cloud/spanner');
+ * const spanner = new Spanner();
+ *
+ * const instance = spanner.instance('my-instance');
+ * const database = instance.database('my-database');
+ * const transaction = database.batchTransaction(identifier);
+ *
+ * transaction.createReadPartitions(options, function(err, partitions) {
+ *   const partition = partitions[0];
+ *
+ *   transaction
+ *     .executeStream(partition)
+ *     .on('error', function(err) {})
+ *     .on('data', function(row) {
+ *       // row = [
+ *       //   {
+ *       //     name: 'SingerId',
+ *       //     value: '1'
+ *       //   },
+ *       //   {
+ *       //     name: 'Name',
+ *       //     value: 'Eddie Wilson'
+ *       //   }
+ *       // ]
+ *     })
+ *     .on('end', function() {
+ *       // All results retrieved
+ *     });
+ * });
+ */
+BatchTransaction.prototype.executeStream = function(partition) {
```

Comment:
@stephenplusplus the naming here was actually at the request of the Spanner team.

Reply:
@vkedia -- we have existing methods named `run` and `runStream`. To deviate and have `run` here and `execute` there is probably not good. Should we be consistent for now, then rename the existing `run` and `runStream` to `execute` and `executeStream` for 2.0?

Reply:
@stephenplusplus it should probably be noted that execute doesn't just execute a query, but instead a partition that could be for a query or a read request. I think its fine to call it execute here, because its not really the same as run/runStream.

Reply:
Alright, whatever you guys think.